.travis.yml

```diff
@@ -1,63 +1,47 @@
 language: python
-sudo: required
-# if the https://travis-ci.org/ODM2/ODM2PythonAPI/requests ever says: missing config
-# validate at: http://lint.travis-ci.org/
-python:
-#  - "2.6"
-  - "2.7"
-#  - "3.2"
-#  - "3.3"
-#  - "3.4"
-#  - "3.5"
-#  - "3.5-dev" # 3.5 development branch
-#  - "nightly" # currently points to 3.6-dev
-# command to install dependencies
-cache:
-  directories:
-    - $HOME/.cache/pip/wheels
-    - $HOME/virtualenv/python2.7.9
-addons:
-  apt:
-    packages:
-      - python-lxml
-      - cmake
-      - unixodbc
-      - unixodbc-dev
-      - odbcinst1debian2
-      - odbcinst
-      - freetds-dev
-      - freetds-bin
-      - tdsodbc
-      - libc6
-      - e2fsprogs
-      - mysql-client
-      - libproj-dev
-      - libgeos-dev
-      - libspatialite-dev
-#services:
- # - mysql
-#  - postgresql
-before_script:
-  - python examples/flask/cbi/build_cbi_cache.py
-  - python examples/flask/swis/runserver_swis.py --config=./test/test_swis_config2.cfg --connection=sqlite:///./test/test_swis2.db &
+
+sudo: false
+
+matrix:
+  fast_finish: true
+  include:
+  - python: 2.7
+    env: TEST_TARGET=default
+  - python: 3.5
+    env: TEST_TARGET=default
+  - python: 3.6
+    env: TEST_TARGET=default
+  - python: 3.6
+    env: TEST_TARGET=coding_standards
+  allow_failures:
```

Comment:
The matrix has only one mandatory test for now. The Python 3 tests and coding standards are optional.